src/screens/ResetWallet.js

```diff
@@ -53,8 +61,16 @@ class ResetWallet extends React.Component {
 
   constructor(props) {
     super(props);
-    this.onBackPress = this.props.navigation.getParam('onBackPress', this.props.navigation.goBack);
-    this.hideBackButton = this.props.navigation.getParam('hideBackButton', false);
+    if (this.props.isScreenLocked) {
+      this.onBackPress = this.props.dropResetOnLockScreen;
+    } else {
+      this.onBackPress = this.props.navigation.getParam('onBackPress', this.props.navigation.goBack);
+    }
+
+    this.hideBackButton = false;
+    if (this.props.navigation) {
+      this.hideBackButton = this.props.navigation.getParam('hideBackButton', false);
+    }
```

Comment:
Another suggestion, using optional chaining:

```suggestion
    this.hideBackButton = this.props.navigation?.getParam('hideBackButton', false) ?? false
```